app/apollo/models/channel.schema.js

```diff
@@ -57,6 +57,9 @@ const ChannelSchema = new mongoose.Schema({
       description: {
         type: String,
       },
+      location: {
+        type: String,
+      },
```

Comment:
It's necessary to define it so that existing data can be loaded and processed.  This was the true underlying bug that was introduced -- when 'location' was removed from the mongoose schema, it meant that mongoose could no longer load and save data in the original format.  When a Channel has multiple Versions and one is deleted, the channel resolver will splice the deleted Version out of the Channel's array and then try to save the update.  The save would fail with a CastError in mongoose when the remaining array elemnt had the location attribute still.  So even though no _current_ code is populating this field, it's necessary to keep it as part of the schema so that existing data can be processed.

Reply:
This looks like a data issue problem - we need to clean the database properly rather than re-introduce data schema elements that are not used anywhere in the code.

Reply:
Requiring data cleanup rather than maintaining backwards compatibility isn't a viable option (and there isn't a data cleanup mechanism defined in Razeedash-api).

Reply:
Backward compatibility was not a requirement when we moved to new way of storing resources, and so removing location field - Channels.versions[...].location - was should have been part of the conversion - https://github.ibm.com/alchemy-containers/satellite-config/issues/1175 - was it not implemented?

Reply:
Backwards compatibility is always a requirement, even if not explicitly stated :-).  Clean up the location field in 1175 was deferred as location removal resulted in breakage in multiple places, such as the cloud CLI and consumers such as https://github.com/IBM/satcon-client-go/.  The location was restored to the graphql schema to avoid this backwards compatibility gap around June 4 ([link](https://github.com/razee-io/Razeedash-api/pull/910)), the current problem arose as a gap because the mongoose schema was also not restored at the same time (while common in real-world usage, it requires very specific steps to recreate the problem [as noted below](https://github.com/razee-io/Razeedash-api/pull/932#issuecomment-876647764))

Reply:
That's correct, Graphql location removal resulted in breakage, but database location field is not required for GraphQl query to return empty location in the result. The above mentioned compatibility was fixed by adding back GraphQl location field. I think the current gap is not because "the mongoose schema was also not restored at the same time", but rather because the location field was not deleted at the same time. If that's right then, now is the good time to go ahead with removing Channels.versions[...].location field.